src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/GeneralizedCoastlineCheck.java

```diff
@@ -57,10 +61,8 @@ public GeneralizedCoastlineCheck(final Configuration configuration)
     @Override
     public boolean validCheckForObject(final AtlasObject object)
```

Comment:
I'm going to offer a suggestion for slight restructuring here to reduce method count and consolidate common control flows to make things a little clearer-- under validCheckForObject, we could have
 ```final Predicate<Relation> memberIsCoastline = relation -> isCoastline(relation);
final Predicate<Relation> memberIsSourcePGS = relation -> isSourcePGS(relation);

return object instanceof LineItem && ((isCoastline(object) && isSourcePGS(object))
                  || (hasRelationMembers(object, memberIsCoastline) && hasRelationMembers(object, memberIsSourcePGS))); 
 ```

Then we could get rid of the wrappers hasCoastlineRelationMembers() and hasSourcePGSRelationMembers().